### PR TITLE
feat: Add synchronizer bhop trainer

### PR DESCRIPTION
**Pull request by tsa96**
I'm the world's biggest idiot and merged Peen's branch by accident. Reverted it, resubmitting the PR. Has my approval, needs 1 more.
Original message:

 adds bhop trainer with 4 visualization modes and settings entries for feedback customization

fixup

### CODE REVIEW

File: scripts/hud/synchronizer.js [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+		return this.getColorStringFromArray(this.lerpColorArrays(colorA, colorB, alpha));
+	}
+
+	static lerpColorArrays(A, B, alpha) {
+		return A.map((Ai, i) => Ai + alpha * (B[i] - Ai));
+	}
+
+	/*
+	 * Sets display mode
+	 */
```
Comment (Gocnak) [inline scripts/hud/synchronizer.js:290]: Avoid "duh" comments; this is just repeating what the function name should already be saying.

File: scripts/hud/synchronizer.js [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+				this.panels.segments[3].style.width = 50 + '%';
+				this.panels.segments.forEach(
+					(segment, i) => (segment.style.backgroundColor = i !== 1 ? altColor : color)
+				);
+
+				// show needle
+				this.panels.needle.style.visibility = 'visible';
+				break;
+			case 4: // "Synchronizer"
+				this.panels.container.style.flowChildren = 'right';
```
Comment (Gocnak) [inline scripts/hud/synchronizer.js:175]: Do these style changes need to happen every `onUpdate`? Shouldn't this happen only on convar change?

**Review by Gocnak — CHANGES_REQUESTED** [review]

File: scripts/hud/synchronizer.js [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+	 */
+	static setSynchroMode() {
+		const newMode = GameInterfaceAPI.GetSettingFloat('mom_hud_synchro_mode');
+
+		this.mom_hud_synchro_mode = newMode === null ? 0 : newMode;
+		if (this.mom_hud_synchro_mode === 0) {
+			this.panels.container.style.visibility = 'collapse';
+		} else {
+			this.panels.container.style.visibility = 'visible';
+		}
```
[the remark this answers is not in the document]
Comment (PeenScreeker) [reply scripts/hud/synchronizer.js]: I was able to shuffle the style around in `synchronizer.scss` to make this work with a ConVarEnabler

File: scripts/hud/synchronizer.js [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+	}
+
+	/*
+	 * Sets colorize mode
+	 */
+	static setSynchroColorMode() {
+		const newColorMode = GameInterfaceAPI.GetSettingFloat('mom_hud_synchro_color_enable');
```
[the remark this answers is not in the document]
Comment (PeenScreeker) [reply scripts/hud/synchronizer.js]: fixed

File: scripts/hud/synchronizer.js (hunk 2 above)
Comment (PeenScreeker) [reply scripts/hud/synchronizer.js]: style that doesn't change every frame is now handled in `setSynchroMode`

**Comment by PeenScreeker** [comment]
Partially fulfills open issue #1629. Detailed stats text display outstanding.

**Review by Gocnak — APPROVED** [review]

**Comment by tsa96** [comment]
I'm fine with this still, just can't approve since it's my issue. @Gocnak can you force approve it? Otherwise Peen you could just approve in my place I guess.

**Pull request merged**